### ISSUE TITLE
BUG: Anchor links in Table of Contents don't scroll content in visually correct way

### ISSUE THREAD

**Issue by matthewpereira** [pivotal/mkdocs-pivotal-theme#6]
**TLDR: Anchor links aren't working that well, we can fix them with CSS or make a smarter long term, easier to maintain fix with a better div/position strategy. WDYT?**

I was looking at the anchor link behavior of each section's table of contents, and noticed that headings/sections aren't actually scrolled into view when you click the TOC to navigate:

![1](https://user-images.githubusercontent.com/6372810/65328977-90eb8c00-db85-11e9-9b0a-6f080280ac68.gif)

Seems like the new theme doesn't seem to take into account that the main container is ~153px from the top of the viewport now, so the first bit of each section is hidden from view. It's technically working the way it should, but isn't visually correct because of the way the headers are positioned on top:

![2](https://user-images.githubusercontent.com/6372810/65329041-ad87c400-db85-11e9-9bd9-9375e92b45ef.gif)

(I added smooth scrolling to make it a little easier to see what's going on)

There are a bunch of different hacky ways to fix this (eg, adding padding and negative margin, so that the browser's positioning of the headers being used as anchors doesn't need to change, and the user doesn't see a visual change to the page's spacing, but the text content is aligned correctly):

![4](https://user-images.githubusercontent.com/6372810/65329114-d314cd80-db85-11e9-9001-f51f18dfb30e.gif)

...But the better way would probably be to use flexbox to create a better scroll-container arrangement that doesn't have main container content overflowing underneath the headers.

Before I started messing with the code, I wanted to check with you fine folks about whether you'd prefer one strategy or another (there are plenty of ways to handle this). The hacky solutions are just a few CSS line changes, but the more elegant and technically sound solutions probably mean a fair bit more `position:` rethinking, and probably changes to the DOM templates as well. I imagine this can be tricky if we're inheriting a lot of stuff from the Material theme.

Happy to get my hands dirty doing the actual work one way or another, let me know what you think!
